parity/params.rs

```diff
@@ -100,20 +99,13 @@ impl Pruning {
 
 	fn find_best_db(dirs: &Directories, genesis_hash: H256, fork_name: Option<&String>) -> Algorithm {
 		let mut algo_types = Algorithm::all_types();
-
-		// if all dbs have the same latest era, the last element is the default one
+		// if all dbs have the same modification time, the last element is the default one
 		algo_types.push(Algorithm::default());
 
 		algo_types.into_iter().max_by_key(|i| {
-			let client_path = dirs.client_path(genesis_hash, fork_name, *i);
-			let config = DatabaseConfig::with_columns(client::DB_NO_OF_COLUMNS);
-			let db = match Database::open(&config, client_path.to_str().unwrap()) {
-				Ok(db) => db,
-				Err(_) => return 0,
-			};
-			let db = journaldb::new(Arc::new(db), *i, client::DB_COL_STATE);
-			trace!(target: "parity", "Looking for best DB: {} at {:?}", i, db.latest_era());
-			db.latest_era().unwrap_or(0)
+			let mut client_path = dirs.client_path(genesis_hash, fork_name, *i);
+			client_path.push("CURRENT");
+			fs::metadata(&client_path).and_then(|m| m.modified()).ok()
 		}).unwrap()
```

Comment:
Will fail if all paths result in `Err -> None` should probably be `unwrap_or_default`